lec13-time-series.Rmd

```diff
@@ -0,0 +1,940 @@
+spring. $dx/dt$ is the rate of change of position, and $dv/dt$ is the rate of change of velocity. 
+
+$$\frac{dx}{dt} = v$$
+
+$$\frac{dv}{dt} = -\omega^2 x$$
+
+We want to draw a **vector field** in two dimensions to show how the system will evolve through time.
+Our axes are $x$ and $v$. It doesn't matter which one is the vertical axis and which one is
+horizontal, but let's put $x$ on the x-axis and $v$ on the y-axis. These variables aren't 
+population sizes, so they can realistically be negative and non-integers. 
+
+Next, we choose some pairs of $x$ and $v$ and evaluate both $dx/dt$ and $dv/dt$, drawing an
+arrow at $(x,v)$ in the direction of the derivative. Let's also set $\omega = 1$ for convenience.
+Let's start with $(0,0)$: both derivatives are 0, so there's no arrow. For $(0,1)$, $dx/dt = 1$
+and $dv/dt = 0$, so we draw an arrow pointing horizontally in the $+x$ direction. Similarly for
+$(1,0)$. For $(1,1)$, we draw an arrow that goes $1$ unit in the $x$ direction and $-1$ unit in
+the $y$ direction. As we keep filling in this grid, we start to get a picture of how the system
+can behave for different starting points of $x$ and $v$. 
+
+This is analogous to plotting $dN/dt$ vs. $N$ and drawing arrows to the right where $dN/dt$ is
+positive and arrows to the left when it's negative, except that now we have to consider that 
+the rate of change is affecting both variables at the same time. 
+
+#### Challenge
+
+Define a function called `mass_spring` that uses the format required by `ode` to calculate 
+and return $dx/dt$ and $dv/dt$. We will use this function later to plot a phase portrait in R.
+Remember that the arguments for your function must be called `t`, `y`, and `parameters`. 
+
+```{r, eval=FALSE, echo=FALSE}
+# Challenge solution
+
+mass_spring <- function(t, y, parameters) {
+  # calculates dx/dt and dy/dt for a predator-prey model
+  
+  # t: time at which to evaluate derivatives
+  # y: vector of system variables (c(x, v))
+  # parameters: vector of model parameters (c(omega))
+    
+  # the state vector y has two elements because we have two variables
+  x <- y[1] # position
+  v <- y[2] # velocity
+  
+  omega <- parameters[1]
+    
+  # calculate rate of change
+  dx <- v
+  dv <- -omega^2 * x
+  
+  # return rate of change
+  return(list(c(dx, dv)))
+}
+```
+
+### Using `phaseR` to draw phase portraits in R
+
+`phaseR` is a package specifically for drawing phase portraits. It can automatically calculate 
+and plot things like trajectories and vector fields. 
+
+```{r}
+library(phaseR)
+```
+
+```{r, eval=FALSE}
+?phaseR
+```
+
+```{r, eval=FALSE}
+?flowField # the nice thing about this packages is that it takes functions in the same format as `ode`
+```
+
+Let's plot a vector field and some trajectories for the predator-prey model.
+
+```{r}
+# plot vector field: on a grid of points, plot an arrow in the direction of dx/dt and dy/dt
+pp_flowField <- flowField(predator_prey, x.lim = c(0, 10), y.lim = c(0, 10),
```

Comment:
`xlim = c(0, 10), ylim = c(0, 10)` - an update to `phaseR` changed the names of these arguments, which is what's breaking Travis (and also breaking on my end)